azuredevops/internal/service/serviceendpoint/commons.go

```diff
@@ -138,6 +139,42 @@ func deleteServiceEndpoint(clients *client.AggregatedClient, projectID *uuid.UUI
 	return nil
 }
 
+func validateServiceEndpoint(clients *client.AggregatedClient, endpoint *serviceendpoint.ServiceEndpoint, serviceEndpointID *string, maxRetries, retryInterval int) error {
+	reqArgs := serviceendpoint.ExecuteServiceEndpointRequestArgs{
+		ServiceEndpointRequest: &serviceendpoint.ServiceEndpointRequest{
+			DataSourceDetails: &serviceendpoint.DataSourceDetails{
+				DataSourceName: converter.String("TestConnection"),
+			},
+			ResultTransformationDetails: &serviceendpoint.ResultTransformationDetails{},
+			ServiceEndpointDetails: &serviceendpoint.ServiceEndpointDetails{
+				Data:          endpoint.Data,
+				Authorization: endpoint.Authorization,
+				Url:           endpoint.Url,
+				Type:          endpoint.Type,
+			},
+		},
+		Project:    converter.String((*endpoint.ServiceEndpointProjectReferences)[0].ProjectReference.Id.String()),
+		EndpointId: serviceEndpointID,
+	}
+
+	var reqResult *serviceendpoint.ServiceEndpointRequestResult
+	var err error
+	log.Printf(fmt.Sprintf(":: serviceendpoint/commons.go :: %s :: Initiating validation", *endpoint.Name))
+	for i := 0; i < maxRetries; i++ {
```

Comment:
can we replace the retry code with `resource.RetryContext()`, a  simple retry util tool